### ISSUE TITLE
Allow search and aggregations on key/value fields

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#20603]
When we re-implemented indexing on key/value fields to prevent ES mapping explosions , we lost the ability to perform aggregations using the values in these fields.   Elasticsearch itself has implemented the idea of a "Flattened" field, which flattens a nested field into a single large json object that is searchable by key value.  The issue with ES Flattened fields is that it is a $$ only feature.

We can implement a very similar strategy with our current key/value fields by adding one more keyword column, e.g. `value__flat` that would map to `$key__$value` which would allow us to perform similar operations.

related to: #18446

**Comment by wezell** [comment]
```
{
	"query": {
		"query_string": {
			"query": "product.specifications1.key_value:size_*"
		}
	},
    "aggs" : {

        "tag" : {
            
            "terms" : {
                "field" : "product.specifications1.key_value",
                "size" : 100  
            }
        }
    },
	
	"size": 10
}
```

**Comment by wezell** [comment]
Here is an example query that pulls all files of mimeType jpeg (using the metadata):
```
{
    "query" : 
    { 
        "query_string" : 
        {
            "query" : "+image.metadata.key_value:contenttype_image\\/jpeg"
        } 
    },
    "size":10,
    "from":0
    
}
```

**Comment by nollymar** [comment]
**Internal QA:** Passed

**Comment by nollymar** [comment]
**Note to QA:** A full reindex is required

**Comment by bryanboza** [comment]
Fixed, tested on release-21.07, after the changes now we can do that queries without problems
